use stale query while creating repeaters

calellowitz opened the pull request:
## Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
https://dimagi-dev.atlassian.net/browse/SAAS-12077
While creating repeaters, use a stale query so that the read does not block on background reindexing activity.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story
<!--
Describe any other pieces to the safety story including
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->
The only effect of this is that the list of repeaters may be slightly out of date when creating new records, which means it might take a bit longer for newly created repeaters to begin receiving forwarded data.

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

calellowitz committed:
use stale query while creating repeaters

dannyroberts commented:
Looks like it doesn't like `stale=False`, and that for that case it looks like other places omit the parameter altogether. I'll add a fix here and take over this PR to get it through for tonight

dannyroberts committed:
Omit stale param rather than setting to False

dannyroberts committed:
Fix cache invalidation for stale repeaters query

This makes Repeater saves, deletes, etc. eagerly
wait for the view to refresh before then
invalidating the caches for Repater.by_domain with stale_query=True.
This makes stale query actually entirely consistent
but possibly for my inability to deeply understand race conditions.
It also makes tests pass. Because the view is refreshed before
the cache is invalidated, a Repeater save should propagate
near-immediately–however long it takes to refresh the view
and invalidate the cache.

dannyroberts commented:
I tested on staging (though I could have done it locally) that running

```
while True:
    ...:     print(FormRepeater.by_domain('droberts', stale_query=True))
    ...:     time.sleep(.1)
```
in the shell, when I add and remove FormRepeaters from the UI, it propagates instantly.

The only other consideration I wanted to call out specifically is that in my last commit, I essentially make saving changes to a Repeater (but not a repeat record) update couch views synchronously whereas normally it would be asynchronous. Since Repeaters change very rarely and a list of repeaters for a domain is fetched all the time, and one of them has to be synchronous in order to achieve consistency, it made sense to shift it from the thing that happens all the time to the thing that happens rarely. I don't expect this to cause any noticeable change, but if it did, it would be in the form of it taking longer to save a change to a Data Forwarder configuration at peak times of the day, which is a great tradeoff for form submissions being decoupled from repeater performance.
